Jenkinsfile.upstream-operator-sync

```diff
@@ -0,0 +1,154 @@
+                        cd version/ && TAG_OPERATOR=\$(grep -m 1 'Version =' version.go) && TAG_OPERATOR=\$(echo \${TAG_OPERATOR#*=} | tr -d '"')
+                        ${CONTAINER_ENGINE} tag quay.io/kiegroup/rhpam-kogito-operator:\${TAG_OPERATOR} ${OPENSHIFT_REGISTRY}/openshift/rhpam-kogito-operator:${operatorTagVersion}
+                        ${CONTAINER_ENGINE} push ${CONTAINER_TLS_OPTIONS} ${OPENSHIFT_REGISTRY}/openshift/rhpam-kogito-operator:${operatorTagVersion}
+                    """
+                }
+            }
+        }
+
+        stage('Run BDD tests') {
+            when {
+                expression {
+                    return goModModified
+                }
+            }
+            options {
+                // Lock on OCP API as we want only one run at a time
+                lock("BDD tests on OCP API ${getLockOpenshiftApi()}")
+            }
+            stages {
+                stage('Running smoke tests') {
+                    steps {
+                        withCredentials([string(credentialsId: "${env.OPENSHIFT_REGISTRY_KEY}" , variable: 'OPENSHIFT_REGISTRY')]) {
+                            // Run just smoke tests to verify basic operator functionality
+                            // Using community example images for now
+                            sh """
+                                make run-tests tags='${env.BDD_TEST_TAGS}' concurrent=5 ${getBDDParameters('always', true)}
+                            """
+                        }
+                    }
+                    post {
+                        always {
+                            archiveArtifacts artifacts: 'test/logs/**/*.log', allowEmptyArchive: true
+                            junit testResults: 'test/logs/**/junit.xml', allowEmptyResults: true
+                        }
+                    }
+                }
+            }
+        }
+    }
+    post {
+        always {
+            cleanWs()
+        }
+    }
+}
+
+// Special method to get the Openshift API in the lock because env is not accessible yet
+void getLockOpenshiftApi() {
+    withCredentials([string(credentialsId: 'OPENSHIFT_API', variable: 'OPENSHIFT_API')]) {
+        return env.OPENSHIFT_API
+    }
+}
+
+String getBDDParameters(String image_cache_mode, boolean runtime_app_registry_internal=false) {
+    testParamsMap = [:]
+
+    testParamsMap["load_default_config"] = true
+    testParamsMap["ci"] = "jenkins"
+    testParamsMap["load_factor"] = 3
+    testParamsMap['disable_maven_native_build_container'] = true
+
+    testParamsMap["operator_image"] = "${OPENSHIFT_REGISTRY}/openshift/rhpam-kogito-operator"
+    testParamsMap["operator_tag"] = "upstream-sync"
+    // Fails with "groovy.lang.MissingPropertyException: No such property: operatorTagVersion for class: groovy.lang.Binding"
+    // testParamsMap["operator_tag"] = operatorTagVersion
```

Comment:
When I uncomment the line I get the issue listed in the comment above. Any idea why or what do I do wrong here?

Reply:
When defining the type in `String operatorTagVersion = "upstream-sync"`, you make local to the script and it is not accessible in a method
you should just define as `operatorTagVersion = "upstream-sync"`, it will make it global